src/commands/command.ts

```diff
@@ -58,6 +55,25 @@ export default abstract class Command {
     return false;
   }
 
+  /** Filters the given command array by the provided role. */
+  public static filterByRole(commands: Command[], role: UserRole): Command[] {
+    return commands.filter((cmd) => {
+      switch (cmd.role) {
+        case UserRole.OWNER:
+          // Owner commands can only be executed by owners
+          return role === UserRole.OWNER;
+        case UserRole.ADMIN:
+          // Admin commands can only be executed by owners and admins
+          return role === UserRole.OWNER || role === UserRole.ADMIN;
+        case UserRole.USER:
+          // User commands can be executed by anyone
+          return true;
+        default:
+          return false;
+      }
+    });
+  }
```

Comment:
I think this makes sense here as well. Tbh, I think most static methods shouldn't be part of a class anyway but since we have that style everywhere I'm fine with this being here